### PR TITLE
resource/aws_emr_cluster: Add Kerberos support

### LINKED ISSUE #3477: aws_emr_cluster enhancement to support KerberosAttributes
### Terraform Version
0.10.8
aws_provider 1.9.0

### Affected Resource(s)
- aws_emr_cluster

### Expected Behavior
Support the following attributes of the RunJobFlow API documented at https://docs.aws.amazon.com/emr/latest/APIReference/API_KerberosAttributes.html
```
   "KerberosAttributes": { 
      "ADDomainJoinPassword": "string",
      "ADDomainJoinUser": "string",
      "CrossRealmTrustPrincipalPassword": "string",
      "KdcAdminPassword": "string",
      "Realm": "string"
   }
```

### PR DESCRIPTION
**Pull request by bflad**
Closes #3477 

This will require a lot more time to setup a proper Active Directory acceptance test.

```
make testacc TEST=./aws TESTARGS='-run=TestAccAWSEMRCluster_Kerberos_ClusterDedicatedKdc'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSEMRCluster_Kerberos_ClusterDedicatedKdc -timeout 120m
=== RUN   TestAccAWSEMRCluster_Kerberos_ClusterDedicatedKdc
--- PASS: TestAccAWSEMRCluster_Kerberos_ClusterDedicatedKdc (406.15s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	406.202s
```

### CODE REVIEW

**Comment by FireballDWF** [comment]
If you can give me a test build which includes this code change, I can run a test which supplies the same parameters as the cluster I successfully created via cli and got working with with AD server.

**Comment by bflad** [comment]
This was (manually) verified in an environment with Active Directory and the relevant AD attributes.

File: aws/resource_aws_emr_cluster.go [hunk 1]
```diff
@@ -532,6 +576,10 @@ func resourceAwsEMRClusterRead(d *schema.ResourceData, meta interface{}) error {
 		log.Printf("[ERR] Error setting EMR Ec2 Attributes: %s", err)
 	}
 
+	if err := d.Set("kerberos_attributes", flattenEmrKerberosAttributes(d, cluster.KerberosAttributes)); err != nil {
+		log.Printf("[ERR] Error setting EMR Kerberos Attributes: %s", err)
```
Comment (paultyng) [inline aws/resource_aws_emr_cluster.go]: I know the others aren't doing it, but it feels weird to me that this doesn't result in an error returning out of the read?

Comment (bflad) [reply aws/resource_aws_emr_cluster.go]: Updated!

**Comment by bflad** [comment]
This has been released in [version 1.11.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1110-march-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by copumpkin** [comment]
Is there a nice way to handle static passwords like these in terraform? The KDC admin password can't really be changed after the fact so it seems pointless to keep it in the state file (that's what happens right now), and makes me pretty uncomfortable to see it sitting there. Would be nice to be able to tag values as "ephemeral" somehow.

**Pull request merged**